sdk/core/azure-core-experimental/azure/core/experimental/transport/_httpx_async.py

```diff
@@ -117,23 +120,43 @@ async def __anext__(self):
         except StopAsyncIteration:
             self.response.internal_response.close()
             raise
+        except httpx.DecodingError as ex:
+            if len(ex.args) > 1:
+                raise DecodeError(ex.args[0]) from ex
+            raise DecodeError("Failed to decode.") from ex
 
 
 class AsyncHttpXTransport(AsyncHttpTransport):
     """Implements a basic async httpx HTTP sender
 
     :keyword httpx.AsyncClient client: HTTPX client to use instead of the default one
+    :keyword bool client_owner: Decide if the client provided by user is owned by this transport. Default to True.
+    :keyword bool use_env_settings: Uses proxy settings from environment. Defaults to True.
     """
 
-    def __init__(self, **kwargs: Any) -> None:
-        self.client = kwargs.get("client", None)
+    def __init__(
+        self,
+        *,
+        client: Optional[httpx.AsyncClient] = None,
+        client_owner: bool = True,
+        use_env_settings: bool = True,
+        **kwargs: Any
+    ) -> None:
+        self.client = client
+        self.connection_config = ConnectionConfiguration(**kwargs)
+        self._client_owner = client_owner
+        self._use_env_settings = use_env_settings
 
     async def open(self) -> None:
         if self.client is None:
-            self.client = httpx.AsyncClient()
+            self.client = httpx.AsyncClient(
+                trust_env=self._use_env_settings,
+                verify=self.connection_config.verify,
+                cert=self.connection_config.cert,
+            )
 
     async def close(self) -> None:
-        if self.client:
+        if self._client_owner and self.client:
             await self.client.aclose()
             self.client = None
 
```

Comment:
yeah I thought about it and left it out because someone would have to do this `transport=HttpXTransport(client_owner=False)` to be in this scenario. 

so you end up getting a None type error as any operation is tried on it Or I check if `self.client is None and self._client_owner == False`  raise a ValueError. I'm thinking the latter might be better to let the user know if want to pass in a client of their own and manage it ( if they havent dont self.client = client bt themselves by the time open is called)